unfurl/src/commonMain/kotlin/me/saket/unfurl/internal/LruCache.kt

```diff
@@ -0,0 +1,152 @@
+package me.saket.unfurl.internal
+
+internal typealias Weigher<K, V> = (K, V?) -> Int
+
+/**
+ * Multiplatform LRU cache implementation.
+ *
+ * Implementation is based on usage of [LinkedHashMap] as a container for the cache and custom
+ * double linked queue to track LRU property.
+ *
+ * [maxSize] - maximum size of the cache, can be anything bytes, number of entries etc. By default is number o entries.
+ * [weigher] - to be called to calculate the estimated size (weight) of the cache entry defined by its [K] and [V].
+ *             By default it returns 1.
+ *
+ * Cache trim performed only on new entry insertion.
+ */
+internal class LruCache<K, V>(
```

Comment:
Was this file copied from somewhere? If so, it should include their code license.

Even better, let's get rid of this class in favor of https://github.com/ReactiveCircus/cache4k?

Reply:
This was from Apollo KMP library since I could not create a pure Kotlin implementation of LinkedHashMap with access order. Kotlin implementation only allows insertion order.
Also, both implementations are not thread-safe.
I did think of using https://github.com/ReactiveCircus/cache4k, but decided against it to keep third-party dependencies to increase maintainability.

Reply:
https://github.com/apollographql/apollo-kotlin/blob/f88b9a0fc7ef169de890651cab3e5f21c78bb3d8/apollo-normalized-cache-common/src/commonMain/kotlin/com/apollographql/apollo3/cache/normalized/internal/LruCache.kt#L20

Reply:
> I did think of using https://github.com/ReactiveCircus/cache4k, but decided against it to keep third-party dependencies to increase maintainability.

Avoiding 3p dependencies in favor of hand-written code is anything but maintainable. The hand-written code will require unit tests, regular maintenance and effort over its lifetime. 3p libraries on the other hand are already battle tested and should always be preferred.